src/app/js/formats/streamgraph/utils.js

```diff
@@ -0,0 +1,204 @@
+
+import differenceBy from 'lodash/differenceBy';
+import * as d3 from 'd3';
+
+// ===========================================================================================
+// getCssToString get the css js object and return it as string
+// ===========================================================================================
+
+export function getCssToString(cssToChange, styles) {
+    const cssString = Object.keys(styles.legend).map(key => {
+        return `${key}: ${styles.legend[key]};`
+    }).join(" ");
+    return (cssString);
+}
+
+// ===========================================================================================
+// zoomFunction will update the zoom on the graph
+// ===========================================================================================
+
+export function zoomFunction() {
+    // create new scale ojects based on event
+    let new_xScale = d3.event.transform.rescaleX(this.xAxisScale);
+    let new_yScale = d3.event.transform.rescaleY(this.yAxisScale);
+
+    // update axes
+    this.gx.call(this.xAxis.scale(new_xScale));
+    this.gyr.call(this.yAxisR.scale(new_yScale));
+    this.gyl.call(this.yAxisL.scale(new_yScale));
+
+    // update streams
+    this.streams.attr("transform", d3.event.transform);
+};
+
+// ===========================================================================================
+// distinctColors : return a list of @count different and distinct colors in hsl hexa
+// ===========================================================================================
+
+export function distinctColors(count) {
+    let colors = [];
+    for (let hue = 0; hue < 360; hue += 360 / count) {
+        colors.push(hslToHex(hue, 90, 50));
+    }
+    return colors;
+}
+
+// ===========================================================================================
+// hslToHex : Convert hsl values color to hex color
+// ===========================================================================================
+
+export function hslToHex(h, s, l) {
```

Comment:
because when I was looking for a method to have different colors around the palette I find this function so I just copy paste it ! But yes, could use d3-color